dymos/examples/cannonball/doc/test_doc_two_phase_cannonball.py

```diff
@@ -12,100 +12,217 @@ class TestTwoPhaseCannonballForDocs(unittest.TestCase):
-        ascent.set_state_options('r', fix_initial=True, fix_final=False)
-        ascent.set_state_options('h', fix_initial=True, fix_final=False)
-        ascent.set_state_options('gam', fix_initial=False, fix_final=True)
-        ascent.set_state_options('v', fix_initial=False, fix_final=False)
-
-        ascent.add_parameter('S', targets=['aero.S'], units='m**2')
-        ascent.add_parameter('mass', targets=['eom.m', 'kinetic_energy.m'], units='kg')
+        # Final flight path angle is fixed (we will set it to zero
+        # so that the phase ends at apogee)
+        ascent.set_time_options(fix_initial=True, duration_bounds=(1, 100),
+                                duration_ref=100, units='s')
+        ascent.set_state_options('r', fix_initial=True, fix_final=False,
+                                 units='m', rate_source='r_dot')
+        ascent.set_state_options('h', fix_initial=True, fix_final=False,
+                                 units='m', rate_source='h_dot')
+        ascent.set_state_options('gam', fix_initial=False, fix_final=True,
+                                 units='rad', rate_source='gam_dot')
+        ascent.set_state_options('v', fix_initial=False, fix_final=False,
+                                 units='m/s', rate_source='v_dot')
+
+        ascent.add_parameter('S', units='m**2', dynamic=False)
+        ascent.add_parameter('m', units='kg', dynamic=False)
 
         # Limit the muzzle energy
-        ascent.add_boundary_constraint('kinetic_energy.ke', loc='initial',
+        ascent.add_boundary_constraint('ke', loc='initial',
                                        upper=400000, lower=0, ref=100000)
 
         # Second Phase (descent)
         transcription = dm.GaussLobatto(num_segments=5, order=3, compressed=True)
-        descent = CannonballPhase(transcription=transcription)
+        descent = dm.Phase(ode_class=CannonballODE, transcription=transcription)
 
         traj.add_phase('descent', descent)
 
-        # All initial states and time are free (they will be linked to the final states of ascent.
-        # Final altitude is fixed (we will set it to zero so that the phase ends at ground impact)
+        # All initial states and time are free, since
+        #    they will be linked to the final states of ascent.
+        # Final altitude is fixed, because we will set
+        #    it to zero so that the phase ends at ground impact)
         descent.set_time_options(initial_bounds=(.5, 100), duration_bounds=(.5, 100),
                                  duration_ref=100, units='s')
-        descent.add_state('r', )
-        descent.add_state('h', fix_initial=False, fix_final=True)
-        descent.add_state('gam', fix_initial=False, fix_final=False)
-        descent.add_state('v', fix_initial=False, fix_final=False)
+        descent.add_state('r', units='m', rate_source='r_dot')
+        descent.add_state('h', fix_initial=False, fix_final=True,
+                          units='m', rate_source='h_dot')
```

Comment:
I guess in this example, we are being explicit about rate_source and units (because it is a more advanced feature to let dymos figure them out for you)?

Reply:
Good point, since we define the ODE in that same example we should probably just tag the state rates.